src/test/json/json_value_test.cpp

```diff
@@ -56,6 +59,54 @@ struct json_value_test : beast::unit_test::suite
         BEAST_EXPECT(test3 != str);
     }
 
+    void
+    test_StaticStringPtr()
+    {
+        // Some of these tests fail due to an apparent bug in VS2019, but
+        // only debug builds.
+        // The first two blocks are only to confirm that the problem is not
+        // specific to StaticString
+        {
+            constexpr char const* testptrlocal = "A local char const*";
+            static const char* a = testptrlocal;
+            const char* b = testptrlocal;
+            log << "a: " << (void*)a << " " << a << std::endl;
+            log << "b: " << (void*)b << " " << b << std::endl;
+
+            BEAST_EXPECT(strcmp(a, b) == 0);
+            BEAST_EXPECT(a == b);
+        }
+
+        {
+            static const char* a = testptrglobal;
+            const char* b = testptrglobal;
+            log << "a: " << (void*)a << " " << a << std::endl;
+            log << "b: " << (void*)b << " " << b << std::endl;
+
+            BEAST_EXPECT(strcmp(a, b) == 0);
+            BEAST_EXPECT(a == b);  // fails in VS2019 debug
+        }
+
+        {
+            constexpr Json::StaticString testlocal{"A local StaticString"};
+            static const char* a = testlocal.c_str();
+            const char* b = testlocal.c_str();
+            BEAST_EXPECT(a == b);
+            BEAST_EXPECT(a == testlocal.c_str());
+            BEAST_EXPECT(testlocal.c_str() == b);
+            BEAST_EXPECT(testlocal.c_str() == testlocal.c_str());
+        }
+
+        {
+            static const char* a = testglobal.c_str();
+            const char* b = testglobal.c_str();
+            BEAST_EXPECT(a == b);                   // fails in VS2019 debug
+            BEAST_EXPECT(a == testglobal.c_str());  // fails in VS2019 debug
```

Comment:
If we expect these tests to fail with VS2019, does that mean we aren't testing with VS2019? (I can't seem to find a test with VS 2019, but haven't scanned too hard yet.)

Reply:
We do not test with VS2019 debug builds.
* Travis has a single VS2019 build, but it's release.
  * https://github.com/ximinez/rippled/blob/github-ci/.travis.yml#L423-L430
  * https://travis-ci.com/github/ripple/rippled/jobs/506462396
* Github has a job for VS2019 debug, but is intentionally skips the test step.
  * https://github.com/ximinez/rippled/blob/github-ci/.github/workflows/windows.yml#L402-L407
  * https://github.com/ripple/rippled/pull/3851/checks?check_run_id=2624210696 (Notice that the "Test" step is skipped.)